Source/ACE.Server/WorldObjects/Portal.cs

```diff
@@ -134,6 +134,77 @@ public override ActivationResult CheckUseRequirements(WorldObject activator)
                     // You are too powerful to interact with that portal!
                     return new ActivationResult(new GameEventWeenieError(player.Session, WeenieError.YouAreTooPowerfulToUsePortal));
                 }
+
+                var playerPkLevel = player.PkLevel;
+
+                if (PropertyManager.GetBool("pk_server").Item)
+                    playerPkLevel = PKLevel.PK;
+                else if (PropertyManager.GetBool("pkl_server").Item)
+                    playerPkLevel = PKLevel.PKLite;
+
+                if ((PortalRestrictions & PortalBitmask.Undef) != 0)
+                {
+                    // Players may not interact with that portal.
+                    return new ActivationResult(new GameEventWeenieError(player.Session, WeenieError.PlayersMayNotUsePortal));
+                }
+
+                if ((PortalRestrictions & PortalBitmask.NoPk) != 0 && (playerPkLevel == PKLevel.PK || player.PlayerKillerStatus == PlayerKillerStatus.PK))
```

Comment:
I would also recommend changing PortalRestrictions to be the proper PortalBitmask enum type, instead of a nullable int. The default should be Undef

These should be using PortalRestrictions.HasFlag(PortalBitmask.NoNpk) etc.

Reply:
```cs
public PortalBitmask PortalRestrictions
        {
            get => (PortalBitmask)(GetProperty(PropertyInt.PortalBitmask) ?? (int)PortalBitmask.Unrestricted);
            set { if (value == PortalBitmask.Undef) RemoveProperty(PropertyInt.PortalBitmask); else SetProperty(PropertyInt.PortalBitmask, (int)value); }
        }
```